osm-live/Readme.md

```diff
@@ -0,0 +1,153 @@
+```
+
+2. **Generate** `_before.obf` and `*_after.obf` for whole world using OsmAnd MapCreator.
+3. Make `*_diff.obf` - where only changed/deleted objects are present. In the `*_diff.obf` we store objects from `*_after.obf` that are different from `*_before.obf` (created or modified).
+5. Split world's `*_diff.obf` into different countries/region files.
+
+## What to do with OSM relations ?
+
+In OsmAnd OBF file we do not save relation directly, but propagate tags from them to way and nodes.<br>
+
+For example, we have relation [`restriction=only_right_turn`](https://www.openstreetmap.org/relation/8085812) and need to include to OSM-live OBF any member of this relation that was changed or the relation itself. In our case these are:
+- [From way/1169124687](https://www.openstreetmap.org/way/1169124687)
+- [Via node/206392306](https://www.openstreetmap.org/node/206392306)
+- [To way/1170508451](https://www.openstreetmap.org/way/1170508451)
+- [Relation ownself](https://www.openstreetmap.org/relation/8085812)
+
+#### Overpass fetching
+If a way or node was changed, we have to retreive their relation(s) with members and store to `*_before.obf` `*_after.obf`:
+```
+    // 2.3 retrieve all relations for changed nodes / ways, so we can propagate proper tags to them
+    (relation(bn.a);.a;) ->.a;
+    (relation(bw.a);.a;) ->.a;
+    // 3. final step make all relations / way / node complete
+    (way(r.a);.a;) ->.a; 
+    (node(r.a);.a;) ->.a;
+```
+
+If the relation changed itself when we retreive members (nodes and ways), after retrieving all member's relations (even if incompleted - without all the members included!!!), the data is stored to `*_before_rel.obf` and `*_after_rel.obf`:
+```
+    // get all relation changed between START - END
+    (
+      relation(changed:\"$START_DATE\",\"$END_DATE\");
+    )->.a;
+    // 1. retrieve all members of changed relation (nodes/ways) to set .b    
+    (way(r.a);) ->.b; 
+    (node(r.a);.b;) ->.b;
+    // 2. complete ways
+    (node(w.b);.b;) ->.b;
+    // 3. find incomplete relations for all members to set .c
+    (relation(bw.b);) ->.c;
+    (relation(bn.b);.c;) ->.c;
+```
+Note. `*_before_rel.obf` and `*_after_rel.obf` - needs only for find nodes/ways with propagated relation tags. Relations can be created/modified/deleted and need to be sure that these changes is correctly processing in all their members.
+
+#### Utility `generate-relation-osm` . Create `*_after_rel_m.osm`
+```
+    echo "### 1. Generate relation osm : $(date -u) . All nodes and ways copy from before_rel to after_rel " &
+    $OSMAND_MAP_CREATOR_PATH/utilities.sh generate-relation-osm \
+            $DATE_DIR/src/${BASENAME}_before_rel.osm.gz $DATE_DIR/src/${BASENAME}_after_rel.osm.gz \
+            $DATE_DIR/src/${BASENAME}_diff.osm.gz ${BASENAME}_after_rel_m.osm.gz
+```
+Files `*_before_rel.osm` and `*_after_rel.osm` can include different sets of relations. And different relations contains different sets of members (nodes/ways)!!!<br>
```

Comment:
Files `*_before_rel.osm` and `*_after_rel.osm` can include different sets of relations based on overpass query.

> And different relations contains different sets of members (nodes/ways)!!!
Why it's important information, looks clear? with !!!

> - Why is it happen?
It's clear based on the example of the query above, no need to explain... 
1, 2 - easily generates different sets of way, node and 3 generates different sets of the relations